hack/vendor.sh

```diff
@@ -1,28 +1,37 @@
 #!/usr/bin/env bash
 
-# This file is just wrapper around vndr (github.com/LK4D4/vndr) tool.
-# For updating dependencies you should change `vendor.conf` file in root of the
-# project. Please refer to https://github.com/LK4D4/vndr/blob/master/README.md for
-# vndr usage.
+# This file is just wrapper around 'go mod vendor' tool.
+# For updating dependencies you should change `vendor.mod` file in root of the
+# project.
 
 set -e
 
-if ! hash vndr; then
-	echo "Please install vndr with \"go get github.com/LK4D4/vndr\" and put it in your \$GOPATH"
-	exit 1
-fi
+cat > go.mod << EOF
+module github.com/moby/moby
+
+go 1.17
+EOF
+
+cat > hack/make/.resources-windows/go.mod << EOF
+module github.com/docker/docker/autogen/winresources/dockerd
+
+go 1.17
+EOF
 
 if [ $# -eq 0 ] || [ "$1" = "archive/tar" ]; then
 	echo "update vendored copy of archive/tar"
 	: "${GO_VERSION:=$(awk -F '[ =]' '$1 == "ARG" && $2 == "GO_VERSION" { print $3; exit }' ./Dockerfile)}"
-	rm -rf vendor/archive
-	mkdir -p ./vendor/archive/tar
+	rm -rf vendor/archive/tar internal/archive/tar
+	mkdir -p internal/archive/tar
 	echo "downloading: https://golang.org/dl/go${GO_VERSION%.0}.src.tar.gz"
 	curl -fsSL "https://golang.org/dl/go${GO_VERSION%.0}.src.tar.gz" \
-		| tar --extract --gzip --directory=vendor/archive/tar --strip-components=4 go/src/archive/tar
-	patch --strip=4 --directory=vendor/archive/tar --input="$PWD/patches/0001-archive-tar-do-not-populate-user-group-names.patch"
+		| tar --extract --gzip --directory=internal/archive/tar --strip-components=4 go/src/archive/tar
+	patch --strip=4 --directory=internal/archive/tar --input="$PWD/patches/0001-archive-tar-do-not-populate-user-group-names.patch"
 fi
 
 if [ $# -eq 0 ] || [ "$1" != "archive/tar" ]; then
-	vndr -whitelist='^archive[/\\]tar' "$@"
+	go mod vendor -modfile vendor.mod
 fi
+
+rm -rf vendor/archive/tar
+cp -a internal/archive/tar vendor/archive/tar
```

Comment:
Looks like it's failing here;

```
cp: cannot create directory 'vendor/archive/tar': No such file or directory
```

```suggestion
mkdir -p
cp -a internal/archive/tar vendor/archive/tar
```

That said, I'm a bit confused, because in this branch, `vendor/archive/tar` doesn't exist (only `internal/archive/tar`), and we don't reference `internal/archive/tar`, or have the `replace` rule anymore?

Should the script be rewritten to use a temp-dir for the git clone, and then move the files? (and don't have the rename `vendor/archive/... -> internal/archive/...` in this branch?